### PR TITLE
fixes aon assignment

### PR DESCRIPTION
**Pull request by janzill**
This fixes AoN assignment by using linear approximation infrastructure. I just noticed it is broken when trying to use it.

### CODE REVIEW

File: aequilibrae/paths/traffic_assignment.py [hunk 1]
```diff
@@ -231,7 +232,7 @@ def set_algorithm(self, algorithm: str):
             raise AttributeError(f"Assignment algorithm not available. Choose from: {','.join(self.all_algorithms)}")
 
         if algo == "all-or-nothing":
```
Comment (pedrocamargo) [inline aequilibrae/paths/traffic_assignment.py:234]:
```suggestion
        if algo in ["all-or-nothing", "msa", "frank-wolfe", "cfw", "bfw"]:
```

File: aequilibrae/paths/traffic_assignment.py [hunk 2]
```diff
@@ -231,7 +232,7 @@ def set_algorithm(self, algorithm: str):
...
-            self.assignment = allOrNothing(self)
+            self.assignment = LinearApproximation(self, algo)
         elif algo in ["msa", "frank-wolfe", "cfw", "bfw"]:
```
Comment (pedrocamargo) [inline aequilibrae/paths/traffic_assignment.py:236]:
```suggestion
```

File: aequilibrae/paths/traffic_assignment.py [hunk 3]
```diff
@@ -231,7 +232,7 @@ def set_algorithm(self, algorithm: str):
             raise AttributeError(f"Assignment algorithm not available. Choose from: {','.join(self.all_algorithms)}")
 
         if algo == "all-or-nothing":
-            self.assignment = allOrNothing(self)
+            self.assignment = LinearApproximation(self, algo)
```
Comment (pedrocamargo) [inline aequilibrae/paths/traffic_assignment.py:235]:
```suggestion
```

**Comment by pedrocamargo** [comment]
Looks good to me.  Happy for you to accept the suggestions or carry on without them.

**Review by pedrocamargo — APPROVED** [review]

**Comment by janzill** [comment]
makes sense, leftover from a different implementation I had going first

**Pull request merged**